### PR TITLE
Added commands to retrieve, update and delete a Lambda

### PR DESCRIPTION
**Pull request by rideam**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     // "noResolve": true,                                /* Disallow 'import's, 'require's or '<reference>'s from expanding the number of files TypeScript should add to a project. */
 
     /* JavaScript Support */
-    // "allowJs": true,                                  /* Allow JavaScript files to be a part of your program. Use the 'checkJS' option to get errors from these files. */
+    "allowJs": true,                                     /* Allow JavaScript files to be a part of your program. Use the 'checkJS' option to get errors from these files. */
```
Comment (ColinFrick) [inline tsconfig.json]: Is this change necessary?

File: src/utils.ts [hunk 2]
```diff
@@ -127,3 +116,20 @@ export const getLocaleFromLocalizedMessageFileName = (path: string): string | un
...
+
+export function toString(item:  String | undefined): string {
+    return (item ?? "").toString();
+}
+
+export function toJson(item: Object | undefined): string {
+    return JSON.stringify(item ?? "", null, 4)
+}
+
+function errorIfFalse(value:  Object | undefined, message: string) {
```
Comment (ColinFrick) [inline src/utils.ts]: Could you change the method name to `errorIfFalsy`?

IMHO `errorIfFalse` indicates a strict check on `false`.

File: package.json [hunk 3]
```diff
@@ -4,10 +4,12 @@
   "description": "FusionAuth CLI",
   "main": "dist/index.js",
   "scripts": {
-    "build": "tsc",
-    "start": "ts-node src/index.ts",
+    "build": "npx tsc && npm run setVersionNumber && npm run makeIndexJsExecutable",
+    "prepublishOnly": "npm run build",
+    "start": "npx ts-node --esm src/index.ts",
```
Comment (ColinFrick) [inline package.json]: Is `npx` required?

Node.js should call the locally installed packages (through `devDependencies`) first, so - if they are installed - global packages should not be called.

File: package.json [hunk 4]
```diff
@@ -4,10 +4,12 @@
...
     "test": "echo \"Error: no test specified\" && exit 1",
-    "prepublishOnly": "npm run build"
+    "makeIndexJsExecutable": "chmod +x ./dist/index.js",
+    "setVersionNumber": "sed -i \"s/versionfrompackagejson/$(jq -r .version package.json)/g\" ./dist/index.js"
```
Comment (ColinFrick) [inline package.json]: I'm unable to build `node-cli` on MacOS:

image

Additionally `sed`, `chmod` and `jq` are not cross-platform. We either need to add these commands to the requirements for Dev, or use a cross-platform solution like:
```json
    "setVersionNumber": "cross-env replace-in-file versionfrompackagejson $npm_package_version ./dist/index.js"
```
Uses [cross-env](https://www.npmjs.com/package/cross-env) and [replace-in-file](https://www.npmjs.com/package/replace-in-file)

File: package.json [hunk 5]
```diff
@@ -4,10 +4,12 @@
...
   "main": "dist/index.js",
   "scripts": {
-    "build": "tsc",
-    "start": "ts-node src/index.ts",
+    "build": "npx tsc && npm run setVersionNumber && npm run makeIndexJsExecutable",
+    "prepublishOnly": "npm run build",
+    "start": "npx ts-node --esm src/index.ts",
     "test": "echo \"Error: no test specified\" && exit 1",
-    "prepublishOnly": "npm run build"
+    "makeIndexJsExecutable": "chmod +x ./dist/index.js",
```
Comment (ColinFrick) [inline package.json]: Seems only necessary if calling `index.js` directly. Will fail when building on Windows.

**Review by ColinFrick — CHANGES_REQUESTED** [review]
Thanks for the PR @rideam 

There are some unused imports in `lambda-delete.ts` and `lambda-update.ts`. Could you clean them up?

**Comment by sixhobbits** [comment]
@ColinFrick - thanks for the feedback - we've implemented the changes. Let me know if you have any more feedback. We'll open a PR to the docs that includes a guide on how to use this new functionality soon.

**Comment by mooreds** [comment]
@ColinFrick is this approved?

**Review by ColinFrick — APPROVED** [review]

**Pull request merged**